### PR TITLE
[TF FE][JAX FE] Support latest TF 2.18, JAX 0.4.35 and NumPy 2.x

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Support TF 2.18 and JAX 0.4.35

**Ticket:** TBD

### CODE REVIEW

**Review by gkrivor — APPROVED** [review]

**Review by akuporos — APPROVED** [review]

**Comment by akuporos** [comment]
Should also TF in the src/bindings/python/constraints.txt be updated as it was in the PR from dependabot?

**Comment by rkazants** [comment]
> Should also TF in the src/bindings/python/constraints.txt be updated as it was in the PR from dependabot?

Great catch, done:)

File: src/bindings/python/constraints.txt [hunk 1]
```diff
@@ -20,7 +20,7 @@ patchelf<=0.17.2.1
 h5py>=3.1.0,<3.13.0
 docopt~=0.6.2
 paddlepaddle==2.6.0
-tensorflow>=1.15.5,<2.18.0
+tensorflow>=1.15.5,<2.19.0
```
Comment (rkazants) [inline src/bindings/python/constraints.txt:23]:
```suggestion
tensorflow>=1.15.5,<2.18.0
```

**Review by mvafin — APPROVED** [review]

File: tests/requirements_tensorflow [hunk 2]
```diff
@@ -6,10 +6,10 @@ pytest-html==4.1.1
 transformers==4.45.1
 # install exact keras version since tensorflow depends and has no upper bound for it
 keras==3.6.0
-tensorflow==2.17.0; platform_system != "Darwin" or platform_machine != "x86_64"
+tensorflow==2.18.0; platform_system != "Darwin" or platform_machine != "x86_64"
 tensorflow==2.16.2; platform_system == "Darwin" and platform_machine == "x86_64"
 # tensorflow-text is not available for both Windows and ARM platforms
 tensorflow-text==2.17.0; platform_system == "Linux" and platform_machine == "x86_64"
```
Comment (rkazants) [inline tests/requirements_tensorflow:12]:
```suggestion
tensorflow-text==2.18.0rc0; platform_system == "Linux" and platform_machine == "x86_64"
```

File: src/frontends/jax/src/op/erfc.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+namespace ov {
+namespace frontend {
+namespace jax {
+namespace op {
+
+OutputVector translate_erfc(const NodeContext& context) {
+    num_inputs_check(context, 1, 1);
+    auto x = context.get_input(0);
+
+    // create y const of the same type as x
```
Comment (rkazants) [inline src/frontends/jax/src/op/erfc.cpp:23]:
```suggestion
    // create const one of the same type as x
```

File: tests/layer_tests/jax_tests/test_erfc.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+        def jax_erfc(x):
+            return jax.lax.erfc(x)
+
+        return jax_erfc, None, 'erfc'
+
+    @pytest.mark.parametrize("input_shape", [[2], [3, 4]])
+    @pytest.mark.parametrize("input_type", [np.float16, np.float32, np.float64])
+    @pytest.mark.nightly
+    @pytest.mark.precommit
```
Comment (rkazants) [inline tests/layer_tests/jax_tests/test_erfc.py:34]:
```suggestion
```

**Pull request merged**